routes/error-tracker.js

```diff
@@ -128,88 +85,60 @@ function handler(req, res, params) {
       return null;
     }
 
-    const stack = standardizeStackTrace(
-      safeDecodeURIComponent(params.s || ''),
-      message
-    );
+    const stack = standardizeStackTrace(stacktrace, message);
     if (ignoreMessageOrException(message, stack)) {
       res.sendStatus(statusCodes.BAD_REQUEST);
       return null;
     }
-
-    const normalizedMessage = /^[A-Z][a-z]+: /.test(message)
-      ? message
-      : `Error: ${message}`;
-    const event = {
-      serviceContext: {
-        service: errorType,
-        version: version,
-      },
-      message: normalizedMessage,
-      context: {
-        httpRequest: {
-          method: req.method,
-          url: req.originalUrl,
-          userAgent: req.get('User-Agent'),
-          referrer: referrer,
-        },
-      },
-    };
-
-    if (req.method === 'POST') {
-      event.context.httpRequest.url += '?' + querystring.stringify(params);
-    }
-
-    const metaData = {
-      labels: {
-        'appengine.googleapis.com/instance_name': process.env.GAE_INSTANCE,
-      },
-      resource: {
-        type: 'gae_app',
-        labels: {
-          module_id: process.env.GAE_SERVICE,
-          version_id: process.env.GAE_VERSION,
-        },
-      },
-      severity: 500, // Error.
-    };
-
     if (!debug) {
       res.sendStatus(statusCodes.ACCEPTED);
     }
 
     return unminify(stack, version)
       .then(stack => {
-        if (stack.length) {
-          event.message = event.message + `\n${stack.join('\n')}`;
+        const reqUrl =
+          req.method === 'POST'
+            ? `${req.originalUrl}?${queryString}`
+            : req.originalUrl;
+        const normalizedMessage = /^[A-Z][a-z]+: /.test(message)
+          ? message
+          : `Error: ${message}`;
+        const event = {
+          serviceContext: {
+            service: logTarget.serviceName,
+            version: logTarget.versionId,
+          },
+          message: [normalizedMessage, ...stack].join('\n'),
+          context: {
+            httpRequest: {
+              method: req.method,
+              url: reqUrl,
+              userAgent: req.get('User-Agent'),
+              referrer: referrer,
+            },
+          },
+        };
+        const logPromise = logEvent(logTarget.log, event);
+
+        // TODO(#102): Investigate if this can ever be set, and remove if not.
+        if (debug) {
```

Comment:
LOL awesome, I'll remove these

Reply:
Update: It looks like the e2e tests rely on this to actually write something to the response, since they don't have access to the return value of the handler itself. I'm not as familiar with Mocha and the work required to extract this and rewrite the tests doesn't seem worth it to me right now, so I'm probably going to just leave it as-is.